### ISSUE TITLE
Building a solution in Visual Studio 2017 will cause an error

### ISSUE THREAD

**Issue by pdwyer54** [4d/4D-Plugin-SDK#18]
I'm fairly new to generating a 4D plugin so it could entirely be a setup thing. I generated a plugin with the wizard. I attached a photo of the settings I used. Please note that I did not add any commands to the plugin. I then open the solution in Visual Studio 2017 and immediately tried to build it without changing anything. I tried Rebuild, Build, and Build full program database file for solution and they will all give the same error, LNK2019 unresolved external symbol PluginMain referenced in function FourDPackex, and LNK2020. LNK2020 always happens when LNK2019 happens so really it's just one error. 

If I remove PluginMain from 4DPluginAPI.c it will actually compile properly, but that is kind of an important call so I need to figure out this error. The solution targets Windows SDK version 10.0.17763.0. Based upon what I searched it seems like PluginMain might be getting double declared, or not declared where it should be, PluginMain is declared in the 4DPluginAPI.h file,  not 4DPlugin.h if that makes a difference. 

![4DWizardSettings](https://user-images.githubusercontent.com/24302494/59614354-9bae0500-90ee-11e9-8d44-ac9310c27fc6.PNG)

![VSError](https://user-images.githubusercontent.com/24302494/59614355-9bae0500-90ee-11e9-8c5d-884227a218c9.PNG)

**Comment by miyako** [comment]
do you have 
```c
void PluginMain(PA_long32 selector, PA_PluginParameters params){
   //...
}
```
in your 4DPlugin.c file?

also, is 4DPlugin.c included in the project (the set of source files to compile) ?

if 4DPlugin.c is configured to compile as C++ code,

4D Plugin API.c should also be configure to compile in the same language.

**Comment by pdwyer54** [comment]
There is a PluginMain method in 4DPlugin.cpp, there is no .c file as only a .cpp file was generated. I attached a photo of what was generated when using the wizard. I did not add or remove any files.

![image](https://user-images.githubusercontent.com/24302494/59694649-fc087980-91b6-11e9-9443-60136d32561d.png)


I kind of assume this has something to do with the fact that the 4DPlugin file is .cpp and the 4DPluginAPI file is .c

I am unsure how to check if my 4DPlugin file is included in the project, I kind of assumed because it is on the root level of the solution but if there is a way to check this I am happy to hear it. I can confirm that if I have an error in that file it does show up when I attempt to build.

So this last thing is probably the closest to what the issue is (I think). If they both compile in C I see the same error. But if I try to compile both of them in C++ 4DPluginAPI.c has a bunch of errors:

![image](https://user-images.githubusercontent.com/24302494/59696185-88b43700-91b9-11e9-9e61-31b60c582da8.png)

I assume this has something to do with casting in C vs C++. If this is the issue I can probably go through and attempt to fix all of those.
For additional references:
4DPluginAPI.c 
![image](https://user-images.githubusercontent.com/24302494/59696341-cf099600-91b9-11e9-91bd-f962ad67ad65.png)

4DPluginAPI.h
![image](https://user-images.githubusercontent.com/24302494/59696386-e21c6600-91b9-11e9-9c8c-af1aedda59a6.png)

4DPlugin.cpp
![image](https://user-images.githubusercontent.com/24302494/59696432-f8c2bd00-91b9-11e9-8b5a-44d6526c21fa.png)

**Comment by pdwyer54** [comment]
I put both the 4DPlugin.cpp and the 4DPluginAPI.c to be compiled as C++, corrected all the casting errors and still get the same error.

**Comment by pdwyer54** [comment]
We figured out this issue, it was an issue with the file encoding for 4DPlugin.cpp and 4DPlugin.h. in case it is run into in the future, if anyone runs into this again here is the solution:
1. Right click on the 4DPlugin.cpp file, select Analyze -> Run Code Analysis on this file. If the output if this mentions something about the encoding being wrong, then this is probably the issue.
2. Open the 4DPlugin.cpp file, then go to File -> Save 4DPlugin.cpp as...
3. When the save window opens (hopefully it will default to where it was), click the dropdown arrow on the save button and select Save with Encoding...
4.  Confirm you want to replace the existing file.
5. In the Advance Save Options that appears we changed Line endings to be Unix, and selected Ok.

Repeat with 4DPlugin.h

This solves this issue, it seems that the wizard encodes these files with a Mac setting so Windows does not know how to read it. I appreciate the help!
